### PR TITLE
[fix] : Removed extension for GameStateStream instead used StockMapStream

### PR DESCRIPTION
**Pull request by satyamurti**
Initially I added `isBankruptStream` and `givesDividends` as extension to the `GameStateSteam` , but stockMap is automatically updated by gameStateStream so `stock.isBankrupt` and `stock.givesDividends` is realtime only. This PR is for that change.

### CODE REVIEW

File: lib/streams/transformations.dart [hunk 1]
```diff
@@ -26,17 +25,10 @@ extension StockMapStreamTransformations on Stream<Map<int, Stock>> {
...
-}
-
-extension GameStateStreamTransformations on Stream<GameStateUpdate> {
-  /// Transforms gameStateStream into a distinct game state stream
-  Stream<bool> isBankruptStream(int stockId, bool isBankrupt) => map(
-      (gameState) => gameState.gameState.stockBankruptState.stockId == stockId
-          ? gameState.gameState.stockBankruptState.isBankrupt
-          : isBankrupt).distinct();
+  Stream<bool> isBankruptStream(int stockId, bool isBankrupt) =>
+      map((stocks) => stocks[stockId]?.isBankrupt ?? isBankrupt).distinct();
```
Comment (Sudhindra3000) [inline lib/streams/transformations.dart:29]:
```suggestion
  Stream<bool> isBankruptStream(int stockId) =>
      map((stocks) => stocks[stockId]!.isBankrupt).distinct();
```
Stock map stream is ValueStream. No need of specifying initialValue for isBankrupt

Comment (Sudhindra3000) [reply lib/streams/transformations.dart:29]: Same for givesDividends also

File: lib/streams/transformations.dart [hunk 2]
```diff
@@ -26,17 +25,10 @@ extension StockMapStreamTransformations on Stream<Map<int, Stock>> {
...
 
-  Stream<bool> givesDividents(int stockId, bool givesDividends) => map(
-      (gameState) => gameState.gameState.stockDividendState.stockId == stockId
-          ? gameState.gameState.stockDividendState.givesDividend
-          : givesDividends).distinct();
+  Stream<bool> givesDividents(int stockId, bool givesDividends) =>
```
Comment (Sudhindra3000) [inline lib/streams/transformations.dart:31]:
```suggestion
  Stream<bool> givesDividendsStream(int stockId) =>
```

**Review by Sudhindra3000 — CHANGES_REQUESTED** [review]

File: lib/streams/transformations.dart (hunk 1 above)
Comment (satyamurti) [reply lib/streams/transformations.dart:29]: okay

**Review by Sudhindra3000 — APPROVED** [review]

**Pull request merged**